src/org/thoughtcrime/securesms/ProfileGalleryFragment.java

```diff
@@ -184,39 +188,84 @@ public void onMediaLongClicked(DcMsg mediaRecord) {
+            .setMessage(info_str)
+            .setPositiveButton(android.R.string.ok, null)
+            .create();
+    d.show();
+    try {
+      //noinspection ConstantConditions
+      Linkify.addLinks((TextView) d.findViewById(android.R.id.message), Linkify.WEB_URLS);
+    } catch(NullPointerException e) {
+      e.printStackTrace();
+    }
+  }
+
+  private void handleDeleteMedia(final Set<DcMsg> messageRecords) {
+    int messagesCount = messageRecords.size();
+
+    new AlertDialog.Builder(getActivity())
+            .setMessage(getActivity().getResources().getQuantityString(R.plurals.ask_delete_messages, messagesCount, messagesCount))
+            .setCancelable(true)
+            .setPositiveButton(R.string.delete, (dialog, which) -> {
+                int[] ids = DcMsg.msgSetToIds(messageRecords);
+                dcContext.deleteMsgs(ids);
+                actionMode.finish();
+            })
+            .setNegativeButton(android.R.string.cancel, null)
+            .show();
+  }
+
+  private void handleShowInChat(final DcMsg dcMsg) {
+    Intent intent = new Intent(getContext(), ConversationActivity.class);
+    intent.putExtra(ConversationActivity.CHAT_ID_EXTRA, dcMsg.getChatId());
+    intent.putExtra(ConversationActivity.STARTING_POSITION_EXTRA, DcMsg.getMessagePosition(dcMsg, dcContext));
+    startActivity(intent);
+  }
+
+  private void handleSaveAttachment(final DcMsg message) {
+    SaveAttachmentTask.showWarningDialog(getContext(), (dialogInterface, i) -> {
+        Permissions.with(getActivity())
+                .request(Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.READ_EXTERNAL_STORAGE)
+                .ifNecessary()
+                .withPermanentDenialDialog(getString(R.string.perm_explain_access_to_storage_denied))
+                .onAllGranted(() -> {
+                    SaveAttachmentTask saveTask = new SaveAttachmentTask(getContext());
+                    SaveAttachmentTask.Attachment attachment = new SaveAttachmentTask.Attachment(
+                            Uri.fromFile(message.getFileAsFile()), message.getFilemime(), message.getDateReceived(), message.getFilename());
+                    saveTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR, attachment);
+                    actionMode.finish();
+                })
+                .execute();
     });
-    builder.setNegativeButton(android.R.string.cancel, null);
-    builder.show();
+  }
+
+  private DcMsg getSelectedMessageRecord() {
+    Set<DcMsg> messageRecords = getListAdapter().getSelectedMedia();
+
+    if (messageRecords.size() == 1) return messageRecords.iterator().next();
+    else                            throw new AssertionError();
+  }
+
+  private void setCorrectMenuVisibility(Menu menu) {
+    Set<DcMsg> messageRecords = getListAdapter().getSelectedMedia();
+
+    if (actionMode != null && messageRecords.size() == 0) {
+      actionMode.finish();
+      return;
+    }
+
+    if (messageRecords.size() > 1) {
```

Comment:
you can refactor that: define a boolean variable (e.g. hasMultipleMessages) with the condition `messageRecords.size() > 1` and use the value for the visibility

Reply:
good idea! thanks!